agent/taskresource/cgroup/cgroup.go

```diff
@@ -345,3 +348,13 @@ func (cgroup *CgroupResource) GetCgroupMountPath() string {
 	defer cgroup.lock.RUnlock()
 	return cgroup.cgroupMountPath
 }
+
+// Initialize initializes the resource fileds in cgroup
+func (cgroup *CgroupResource) Initialize(resourceFields *taskresource.ResourceFields) {
+	cgroup.lock.Lock()
+	defer cgroup.lock.Unlock()
+
+	cgroup.initializeResourceStatusToTransitionFunction()
+	cgroup.ioutil = resourceFields.IOUtil
```

Comment:
wait -- is the ioutil mock the only issue causing the panic?

Reply:
No, the error is coming from agent runtime, not the test. Because the ioutil is nil, when agent try to call the ioutil.remove, it panics.